### PR TITLE
Add Idobao Montex RGB (V2)

### PR DESCRIPTION
**Pull request by IBNobody**
## Description

Add IdoBao Montex RGB (V2), an upgraded PCB from the original V1 from @NachoxMacho . 

The V2 added per-key RGB LEDs using RGBLIGHT in addition to the original underbody LED lighting. Other than extending the LED lighting, I also added the RGBLIGHT_SLEEP option to turn off the LEDs when the host PC is asleep. No changes to the keymaps were made at this time.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/idobao/montex/v2/config.h [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+#define RGB_DI_PIN B1
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_SLEEP
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/idobao/montex/v2/config.h:52]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by IBNobody** [comment]
Thanks, fixed.

**Review by fauxpark — APPROVED** [review]

File: keyboards/idobao/montex/v2/rules.mk [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
```
Comment (drashna) [inline keyboards/idobao/montex/v2/rules.mk:18]: Oh, and: 
```suggestion
AUDIO_ENABLE = no           # Audio output

LAYOUTS = numpad_6x5
```

**Comment by vinorodrigues** [comment]
Is this not a duplicate of #15865 that was PR'ed back in Jan 13th by @peepeetee, who was on contract by IDOBAO to do so?

**Comment by IBNobody** [comment]
> Is this not a duplicate of #15865 ?

Yes and no. It's the same kb, but instead of using RGB_MATRIX which was the hold-up on that PR, it uses the original RGBLIGHT which V1 used. This avoids the VIA conflict.

**Comment by IBNobody** [comment]
Worked with @vinorodrigues to update the code to support RGB_MATRIX rather than RGBLIGHT.

**Comment by vinorodrigues** [comment]
I can confirm that this code is functional and tested with the V2 PCB sold both separately and with current "Montex RGB" kits.  PR #15865 is also *(probably)* correct, albeit for the older "Montex RGB" variants that pre-date the V2 PCB's *(thus that folder should be named `v1rgb`)*.

This code also honours existing VIA users by using a different PID so that it does not conflict with Montex (V1) boards.

**Review by vinorodrigues — APPROVED** [review]

File: keyboards/idobao/montex/v2/rules.mk [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/idobao/montex/v2/rules.mk:18]:
```suggestion
```

File: keyboards/idobao/montex/v2/v2.h [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    K50, K51, K52            \
+) { \
+    { K00, K01, K02, K03, K04 }, \
+    { K10, K11, K12, K13, K14 }, \
+    { K20, K21, K22, K23, K24 }, \
+    { K30, K31, K32, K33, KC_NO }, \
+    { K40, K41, K42, K43, KC_NO }, \
+    { K50, K51, K52, K53, KC_NO }  \
+}
+---------------------------------------- */
```
Comment (fauxpark) [inline keyboards/idobao/montex/v2/v2.h:39]:
```suggestion
```

File: keyboards/idobao/montex/v2/config.h [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+/* NB: VENDOR_ID & PRODUCT_ID et.al. moved to `info.json` */
+// #define VENDOR_ID       0x6964  // "id"
+// #define PRODUCT_ID      0x0227  // 0x0027 | 0x02 = v2 designator (for VIA)
+// #define DEVICE_VER      0x0002
+// #define MANUFACTURER    IDOBAO
+// #define PRODUCT         Montex ID27
+
```
Comment (fauxpark) [inline keyboards/idobao/montex/v2/config.h:28]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by vinorodrigues** [comment]
VIA PR is draft/ready for review once this is PR merged. [#1251](https://github.com/the-via/keyboards/pull/1251)

**Review by fauxpark — APPROVED** [review]

**Comment by IBNobody** [comment]
@drashna Is there anything else I need to do for this?

**Comment by drashna** [comment]
Would prefer that the layouts used the `numpad_6x5` community layout.  Eg: 
https://github.com/drashna/qmk_firmware/blob/78fbc54d83d57b830a51bb52cb86daa394d46baf/layouts/default/numpad_6x5/default_numpad_6x5/keymap.c

**Comment by vinorodrigues** [comment]
ok - changing ... gimme a mo

**Comment by vinorodrigues** [comment]
@IBNobody - you can also delete the `default_nug` keymap entirely.  This is redundant / duplicated code and can be done via other means. *(my bad for suggesting it in the first place)*

**Comment by IBNobody** [comment]
> Would prefer that the layouts used the `numpad_6x5` community layout. Eg: https://github.com/drashna/qmk_firmware/blob/78fbc54d83d57b830a51bb52cb86daa394d46baf/layouts/default/numpad_6x5/default_numpad_6x5/keymap.c

Are you requesting that I use that identical keymap, or is it acceptable to use that keymap and then add the needed RGB controls to my keymap?

After looking at the `numpad_6x5` community layout, am I able to propose changes to it? 

The `numpad_6x5` layout was added with the original Montex V1. The only other keyboard using that layout is the [WE27](https://github.com/qmk/qmk_firmware/tree/a8d440e4e043dc8427f17c07631556d0ff3efc20/keyboards/wekey/we27) added 1 month later, and it follows the Montex V2 layout closer than it does the Montex V1 layout. (Function key is in the lower left.) I would like to propose a change to the `numpad_6x5` community layout to follow suit, especially since the Montex V1 is no longer obtainable for new users.

**Comment by IBNobody** [comment]
> @IBNobody - you can also delete the `default_nug` keymap entirely. This is redundant / duplicated code and can be done via other means. _(my bad for suggesting it in the first place)_

I just had the via_nug there. I am removing it.

**Comment by IBNobody** [comment]
> > Would prefer that the layouts used the `numpad_6x5` community layout. Eg: https://github.com/drashna/qmk_firmware/blob/78fbc54d83d57b830a51bb52cb86daa394d46baf/layouts/default/numpad_6x5/default_numpad_6x5/keymap.c
> 
> Are you requesting that I use that identical keymap, or is it acceptable to use that keymap and then add the needed RGB controls to my keymap?
> 
> After looking at the `numpad_6x5` community layout, am I able to propose changes to it?
> 
> The `numpad_6x5` layout was added with the original Montex V1. The only other keyboard using that layout is the [WE27](https://github.com/qmk/qmk_firmware/tree/a8d440e4e043dc8427f17c07631556d0ff3efc20/keyboards/wekey/we27) added 1 month later, and it follows the Montex V2 layout closer than it does the Montex V1 layout. (Function key is in the lower left.) I would like to propose a change to the `numpad_6x5` community layout to follow suit, especially since the Montex V1 is no longer obtainable for new users.

I'm still not 100% sure how to proceed.

File: keyboards/idobao/montex/v2/keymaps/via/config.h [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 6
+
+#if defined(RGB_MATRIX_ENABLE)
+    #define VIA_QMK_RGBLIGHT_ENABLE
+#endif
```
Comment (drashna) [inline keyboards/idobao/montex/v2/keymaps/via/config.h:23]: This won't be needed anymore, as of the end of the month (or if you compile from the develop branch).
```suggestion
```

**Review by drashna — APPROVED** [review]

**Pull request merged**